bin/collect_stats.py

```diff
@@ -95,26 +128,37 @@ def main(args=None):
 
         # condition_name, total_peptide_count
         total_peptide_count = sum(conditions_peptides["condition_peptide_count"])
-        print("Total peptides:", total_peptide_count, file=args.outfile, sep="\t", flush=True)
 
         # condition_name, unique_peptide_count
         unique_peptide_count = len(conditions_peptides)
-        print("Unique peptides:", unique_peptide_count, file=args.outfile, sep="\t", flush=True)
-        print(file=args.outfile)
-
-    # unique peptides across all conditions
-    all_conditions_unqiue_peptide_counts = len(protein_peptide_occs["peptide_id"].drop_duplicates())
-    print(file=args.outfile)
-    print(
-        "Unique peptides across all conditions:",
-        all_conditions_unqiue_peptide_counts,
-        file=args.outfile,
-        sep="\t",
-        flush=True,
-    )
 
-    print("Done!", flush=True)
+        # peptide_id, condition_name, condition_peptide_count, highest_prediction_score, prediction_score_allele_0, prediction_score_allele_1,...prediction_score_allele_n
+        conditions_peptides = conditions_peptides.set_index("peptide_id").join(best_scored_peptides).join(predictions)
```

Comment:
I am not sure I understood all details of the new code, but since it caused quite some memory usage, I have a few questions/suggestions since I am wondering if you really need that many joins:

-  `conditions_peptides`: if the resulting df become too large, you could drop the `condition_name` (anyway the same within this context) and `count` before joining to the predictions
- filter `predictions` directly after reading in for entries that are above the threshold, and drop the `prediction_score` to reduce the size
-  merge/join for each allele to get only peptides that are "binders" and count

Brings me to the next question: you just provide the count of unique binders, right? Maybe that should be stated somewhere more clearly.